### PR TITLE
MET-5193 Fixed behaviors reported from testing

### PR DESCRIPTION
**Pull request by JoanaCMS**

### CODE REVIEW

File: metis-performance-metric/src/main/java/eu/europeana/metis/performance/metric/utilities/CSVUtilities.java [hunk 1]
```diff
@@ -71,7 +71,9 @@ private String turnDatasetIntoCSVRow(WorkflowExecution execution){
             stringBuilderCSVRow.append(", ");
             stringBuilderCSVRow.append(calculateTimeDifference(execution, (AbstractExecutablePlugin<?>)publishPlugin));
             stringBuilderCSVRow.append(", ");
-            stringBuilderCSVRow.append(((AbstractExecutablePlugin<?>) publishPlugin).getExecutionProgress().getProcessedRecords());
```
Comment (jeortizquan) [inline metis-performance-metric/src/main/java/eu/europeana/metis/performance/metric/utilities/CSVUtilities.java:74]: After reviewing I suggest to rewrite the method with this changes.
final added to variables.
added an extra method to reduce the boxing an unboxing of variables.
changed the logging. there were some sonar recommendations.
```
private String turnDatasetIntoCSVRow(WorkflowExecution execution){
        final String datasetId = execution.getDatasetId();
        final SimpleDateFormat simpleDateFormatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        LOGGER.info("Processing dataset with id {}" , datasetId);
        final StringBuilder stringBuilderCSVRow = new StringBuilder();
        stringBuilderCSVRow.append(datasetId);
        stringBuilderCSVRow.append(", ");
        final Optional<AbstractMetisPlugin> optionalPublishPlugin = execution.getMetisPluginWithType(PluginType.PUBLISH);
        if (optionalPublishPlugin.isPresent()){
            final AbstractExecutablePlugin<?> publishPlugin = (AbstractExecutablePlugin<?>) optionalPublishPlugin.get();
            stringBuilderCSVRow.append(simpleDateFormatter.format(publishPlugin.getFinishedDate()));
            stringBuilderCSVRow.append(", ");
            stringBuilderCSVRow.append(calculateTimeDifference(execution, publishPlugin));
            stringBuilderCSVRow.append(", ");
            stringBuilderCSVRow.append(calculatePublishRecords(publishPlugin));
        } else {
            LOGGER.error("Something went wrong when extracting data");
            return "";
        }

        LOGGER.info("Finished processing dataset with id {}" , datasetId);

        return stringBuilderCSVRow.toString();
    }
    
    private int calculatePublishRecords(AbstractExecutablePlugin<?> publishPlugin){
        return publishPlugin.getExecutionProgress().getProcessedRecords() - publishPlugin.getExecutionProgress().getErrors();
    }
```

File: metis-performance-metric/src/main/java/eu/europeana/metis/performance/metric/dao/MongoMetisCoreDao.java [hunk 2]
```diff
@@ -53,7 +54,11 @@ public WorkflowExecutionDao.ResultList<WorkflowExecutionDao.ExecutionDatasetPair
                 Set.of(PluginType.PUBLISH), Date.from(startLocalDateTime.atZone(ZoneId.systemDefault()).toInstant()), endDate, nextPage, 1000);
 
         while (CollectionUtils.isNotEmpty(resultList.getResults())){
-            pairsList.addAll(resultList.getResults());
```
Comment (jeortizquan) [inline metis-performance-metric/src/main/java/eu/europeana/metis/performance/metric/dao/MongoMetisCoreDao.java:56]: My suggestion is to add some `final` modifier.
to 
1) final List<WorkflowExecutionDao.ExecutionDatasetPair> pairsList
2) final List<WorkflowExecutionDao.ExecutionDatasetPair> filteredResult
```
public WorkflowExecutionDao.ResultList<WorkflowExecutionDao.ExecutionDatasetPair> getAllWorkflowsExecutionsOverview(Date startDate, Date endDate){
        final List<WorkflowExecutionDao.ExecutionDatasetPair> pairsList = new ArrayList<>();
        //Make start date two weeks earlier, so we can include more reports based on end date
        final LocalDateTime startLocalDateTime = LocalDateTime.ofInstant(startDate.toInstant(), ZoneId.systemDefault())
                .minusWeeks(2L);
        int nextPage = 0;
        WorkflowExecutionDao.ResultList<WorkflowExecutionDao.ExecutionDatasetPair> resultList = workflowExecutionDao
                .getWorkflowExecutionsOverview(null, Set.of(PluginStatus.FINISHED),
                Set.of(PluginType.PUBLISH), Date.from(startLocalDateTime.atZone(ZoneId.systemDefault()).toInstant()), endDate, nextPage, 1000);

        while (CollectionUtils.isNotEmpty(resultList.getResults())){
            final List<WorkflowExecutionDao.ExecutionDatasetPair> filteredResult = resultList.getResults()
                    .stream()
                    .filter(pair -> isWithinInterval(pair.getExecution().getFinishedDate(), startDate, endDate))
                    .collect(Collectors.toList());
            pairsList.addAll(filteredResult);
            nextPage++;
            resultList = workflowExecutionDao
                    .getWorkflowExecutionsOverview(null, Set.of(PluginStatus.FINISHED),
                            Set.of(PluginType.PUBLISH), Date.from(startLocalDateTime.atZone(ZoneId.systemDefault()).toInstant()), endDate, nextPage, 1000);
        }

        return new WorkflowExecutionDao.ResultList<>(pairsList, true);
    }
```

**Review by jeortizquan — COMMENTED** [review]
Please have a look on the comments suggestions.

File: metis-performance-metric/src/main/java/eu/europeana/metis/performance/metric/utilities/CSVUtilities.java (hunk 1 above)
Comment (JoanaCMS) [reply metis-performance-metric/src/main/java/eu/europeana/metis/performance/metric/utilities/CSVUtilities.java:74]: Fixed

File: metis-performance-metric/src/main/java/eu/europeana/metis/performance/metric/dao/MongoMetisCoreDao.java (hunk 2 above)
Comment (JoanaCMS) [reply metis-performance-metric/src/main/java/eu/europeana/metis/performance/metric/dao/MongoMetisCoreDao.java:56]: Fixed

**Review by jeortizquan — APPROVED** [review]

**Pull request merged**